2-projects/1-todo-list/script.js

```diff
@@ -1,25 +1,88 @@
+  });
+};
+
+const createListElement = (work) => {
+  let list = document.createElement("li");
+  list.innerText = work.task;
+  list.classList.add(
+    "list-group-item",
+    "d-flex",
+    "justify-content-between",
+    "align-items-center"
+  );
+  ulEl.appendChild(list);
+  return list;
+};
+
+const createSpanElement = (list) => {
+  let span = document.createElement("span");
+  span.classList.add("badge", "bg-primary", "rounded-pill");
+  list.appendChild(span);
+  return span;
+};
+
+const createICheckElement = (span, work, list) => {
+  let iElCheck = document.createElement("i");
+  iElCheck.classList.add("fa", "fa-check");
+  iElCheck.ariaHidden = work.completed;
+  //when check sign click
+  iElCheck.addEventListener("click", () => {
+    list.style.textDecoration === "line-through"
+      ? (list.style.textDecoration = "")
+      : (list.style.textDecoration = "line-through");
+  });
+  span.appendChild(iElCheck);
+};
+
+const createITrashElement = (span, work, list) => {
+  let iElTrash = document.createElement("i");
+  iElTrash.classList.add("fa", "fa-trash");
+  iElTrash.ariaHidden = work.completed;
+  //when delete sign click
+  iElTrash.addEventListener("click", () => {
+    ulEl.removeChild(list);
+  });
+  span.appendChild(iElTrash);
+};
+
 populateTodoList(todos);
 
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
+const addNewTodo = (event) => {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
-}
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
-  // Write your code here...
-}
+  let inputEl = document.getElementById("todoInput");
+  let taskTitle = inputEl.value;
+  let newArrayElement = [{ task: taskTitle, completed: false }];
+  console.log(newArrayElement[0]);
+  let list = createListElement(newArrayElement[0]);
+  let span = createSpanElement(list);
+  createICheckElement(span, newArrayElement[0], list);
+  createITrashElement(span, newArrayElement[0], list);
+  inputEl.textContent = "";
+};
+
+//Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+const deleteAllCompletedTodos = (event) => {
+  event.preventDefault();
+  let list = document.querySelectorAll("li");
+  list.forEach((task) => {
+    if (task.style.textDecoration === "line-through") ulEl.removeChild(task);
+  });
+};
```

Comment:
This is as always such amazing code Laleh! You have clearly understood DOM manipulation, and how Javascript interacts with the HTML elements! This stuff is super hard and very complicated, but you have achieved everything! You should be super proud! 🎉  👯  🙌 

However! Because you are such a good coder, I am able to give you some very in-depth feedback. This is not to say anything is wrong, but here is how you can improve. Again, I want you to be the best developer possible!

If you load this to-do page, and do not enter a task and click `Add Todo`, it will save an empty todo. This should not happen! Also, once you save a todo task, the input field dons't reset, and you can keep adding the same todo over and over again. It should reset to null and force the user to add another todo. 

In both cases, you should throw an error message saying something like "Please enter a Todo task to save!" and NOT save the todo. You can do this by writing an `if` statement in your code and the `addNewTodo` function. 

This is what I have coded.

```
const addNewTodo = (event) => {
	// The code below prevents the page from refreshing when we click the 'Add Todo' button.
	event.preventDefault();

	const errorMessage = document.getElementById("errorhandler");
	let inputEl = document.getElementById("todoInput");
	let taskTitle = inputEl.value;

	// if no task, throw error message
	if(!taskTitle || taskTitle.length < 1)
	{
		errorMessage.innerHTML = "You need to write a task!";
		return false;
	}

	// if no errors, remove old error message and set error message as null
	errorMessage.innerHTML = null;

	let newArrayElement = [{ task: taskTitle, completed: false }];
	console.log(newArrayElement[0]);
	let list = createListElement(newArrayElement[0]);
	let span = createSpanElement(list);
	createICheckElement(span, newArrayElement[0], list);
	createITrashElement(span, newArrayElement[0], list);

	// once new to do is saved, reset input fields
	inputEl.value = null;
	inputEl.innerHTML = null;
};
```

And in the index.html I added a `<div id="errorhandler"></div>` on the line directly below the  `<div class="card-header">Add new Todos</div>`

What I am doing here is when a user clicks the "Add todo" button, we run the `addNewTodo` function, BUT if the `taskTitle` is falsy, empty, or has a length under one, we set the `errorMessage`'s innerHTML to display "You need to write a task!". We also return the function to stop it from running. That way we ensure the user has to enter a todo before we save it.

If a user then enters a todo, we can pass this `if` check, remove the error message by doing `errorMessage.innerHTML = null;` and then run the function as normal.

ALSO. Once this function has run and we have added the new todo item, we should reset the state and innerHTML to ensure the users adds a new todo task. We achive this by adding 
```
	// once new to do is saved, reset input fields
	inputEl.value = null;
	inputEl.innerHTML = null;
```
at the end of the function.

We could go even further with this and add another `if` check incase someone adds a todo that has already been added. For example, we have as a default `Wash the dishes` as a todo. If the user then adds another todo of `Wash the dishes` we should check the todo array, and if it is already there, we should THROW AN ERROR of "this todo has already been added!" and RETURN to stop the function from running.

As an extra challenge, do you think you can add this check? I'll give you a hint below

```
     const todoList = []

	// if no task, throw error message
	if(!taskTitle || taskTitle.length < 1)
	{
		errorMessage.innerHTML = "You need to write a task!";
		return false;
	}

	// if to do is already in the array, throw error message and return
	if(todoList.includes....................)
	{
		errorMessage.innerHTML = "This has already been added!";
		return false;
	}

       taskTitle.push......................
```

Again, I am being extremely critical here, but it is just because I want you to improve as much as possible :)

This is still amazing code and you should be very very proud of what you have achieved!!!